snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/dsv/SnomedSimpleTypeRefSetDSVExporter.java

```diff
@@ -956,86 +680,89 @@ private String joinResultsWithDelimiters(Collection<String> results, int max, St
 		// fill the remaining slots with delimiters
 		for (int j = start; j < max; j++) {
 			sb.append(delimiter);
-			if (descriptionAndDescriptionIdExpected) {
+			if (includeComponentId) {
 				sb.append(delimiter);
 			}
 		}
 		return sb.toString();
 	}
 
-	private Map<Long, String> getExportedMemberConceptCDOAndSnomedIds(String refSetId) throws SQLException {
-		Map<Long, String> conceptCDOIds = Maps.newHashMap();
-		PreparedStatement inactivatedCheckerStatement = connection
-				.prepareStatement(SnomedRefSetExporterQueries.SQL_SIMPLE_TYPE_REFSET_DSV_EXPORT_INACTIVATED_REFSET_MEMBER_CONCEPT_QUERY);
-		PreparedStatement statement = connection.prepareStatement(SnomedRefSetExporterQueries.SQL_SIMPLE_TYPE_REFSET_DSV_EXPORT_REFSET_MEMBER_QUERY);
-		statement.setLong(1, Long.valueOf(refSetId));
-		statement.setInt(2, branchId);
-		statement.setInt(3, branchId);
-		statement.setLong(4, branchBase);
-		statement.setLong(5, branchBase);
-		statement.setInt(6, branchId);
-		statement.setInt(7, branchId);
-		statement.setLong(8, branchBase);
-		statement.setLong(9, branchBase);
-		ResultSet resultSet = statement.executeQuery();
-		int i = 1;
-		while (resultSet.absolute(i)) {
-			Long CDOId = resultSet.getLong(1);
-			inactivatedCheckerStatement.setLong(1, CDOId);
-			inactivatedCheckerStatement.setInt(2, branchId);
-			ResultSet inactivatedCheckerResultSet = inactivatedCheckerStatement.executeQuery();
-			
-			/*
-			 * Active refset members are skipped if the referenced concept is inactive!
-			 */
-			if (inactivatedCheckerResultSet.absolute(1) && !inactivatedCheckerResultSet.getBoolean(1)) {
-				i++;
-				continue;
-			}
-			conceptCDOIds.put(resultSet.getLong(1), resultSet.getString(2));
-			i++;
-		}
-		return conceptCDOIds;
+	
+	/**
+	 * Fetch members of the specified refset identifier.
+	 * Don't include inactive members or members that are active BUT their identified component are inactive.
+	 * @param refSetId
+	 * @return
+	 */
+	private SnomedConcepts getReferencedComponentConcepts(String refSetId) {
+		return SnomedRequests.prepareSearchConcept()
+								.all()
+								.filterByActive(true)
+								.filterByEcl(String.format("^%s", refSetId)) // memberOf refsetId
+								.setExpand("pt()")
+								.setLocales(locales)
+								.build(SnomedDatastoreActivator.REPOSITORY_UUID, branchPath.getPath())
+								.execute(getEventBus())
+								.then(concepts -> expandDescriptions(concepts))
```

Comment:
yeah... 
my original expand query looked like this, but when running it, it ran out of workers :D 

![screen shot 2017-05-26 at 18 04 23](https://cloud.githubusercontent.com/assets/1477846/26680795/feeca42e-46da-11e7-9db3-a4638c74d573.png)